pkg/machines/libvirt-dbus.js

```diff
@@ -541,6 +561,32 @@ LIBVIRT_DBUS_PROVIDER = {
         };
     },
 
+    /*
+     * Read properties of a single NodeDevice
+     *
+     * @param NodeDevice object path
+     */
+    GET_NODE_DEVICE({
+        id: objPath,
+        connectionName,
+    }) {
+        let props = {};
+
+        return dispatch => {
+            call(connectionName, objPath, 'org.libvirt.NodeDevice', 'GetXMLDesc', [0], TIMEOUT)
+                    .then(deviceXml => {
+                        let result = xml2js(deviceXml, { compact: true });
+
+                        props.connectionName = connectionName;
+                        props.xml = result;
```

Comment:
That's confusing, as this is now a JS object, not XML. (But of course, let's please try and keep it as XML; a string, or possibly a [DOMParser](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser) object)

Reply:
I will rewrite the PR in this direction, np :)